File: 029_Strings_Character_Arrays_Part_1/7_swap.cpp
#include <iostream>
#include <vector>
#include <algorithm> // for swap
using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int st = 0;
        int en = s.size() - 1;

        // while (st < en) {
        //     swap(s[st], s[en]);
        //     st++;
        //     en--;
        // }

        while (st < en)
        swap(s[st++], s[en--]);

    }
};

int main() {
    vector<char> s = {'h', 'e', 'l', 'l', 'o'};
    Solution obj;
    obj.reverseString(s);

    for (char c : s) {
        cout << c;
    }
    cout << endl;

    return 0;
}

File: 029_Strings_Character_Arrays_Part_1/7_swap.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        st = 0
        en = len(s) - 1
        while st < en:
            s[st], s[en] = s[en], s[st]  # swap
            st += 1
            en -= 1


# Example usage
s = ["h", "e", "l", "l", "o"]
obj = Solution()
obj.reverseString(s)
print("Reversed string:", s)